data-prepper-plugins/sns-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/SNSSinkConfig.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.plugins.sink;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.NotEmpty;
+import jakarta.validation.constraints.NotNull;
+import org.opensearch.dataprepper.model.configuration.PluginModel;
+import org.opensearch.dataprepper.plugins.sink.configuration.AwsAuthenticationOptions;
+
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * sns sink configuration class contains properties, used to read yaml configuration.
+ */
+public class SNSSinkConfig {
+
+    private static final int DEFAULT_CONNECTION_RETRIES = 5;
+
+    private static final int DEFAULT_BATCH_SIZE = 10;
+
+    private static final int DEFAULT_UPLOAD_RETRIES = 5;
+
+    public static final String STS_REGION = "region";
+
+    public static final String STS_ROLE_ARN = "sts_role_arn";
+
+    @JsonProperty("aws")
+    @NotNull
+    @Valid
+    private AwsAuthenticationOptions awsAuthenticationOptions;
+
+    @JsonProperty("topic")
+    @NotNull
+    @NotEmpty
+    private String topicArn;
+
+    @JsonProperty("message_group_id")
+    private String messageGroupId;
+
+    @JsonProperty("codec")
+    @NotNull
+    private PluginModel codec;
+
+    @JsonProperty("dlq")
+    private PluginModel dlq;
+
+    @JsonProperty("dlq_file")
+    private String dlqFile;
+
+    @JsonProperty("batch_size")
+    private int batchSize = DEFAULT_BATCH_SIZE;
+
+    @JsonProperty("message_deduplication_id")
```

Comment:
Missing docs for this option.

Reply:
Thanks @asifsmohammed for your feedback, Sure, it will available in next commit for review.